pkg/cosign/tuf/client.go

```diff
@@ -503,49 +489,175 @@ func rootCacheDir() string {
 		if err != nil {
 			home = ""
 		}
-		return filepath.Join(home, ".sigstore", "root")
+		return filepath.FromSlash(filepath.Join(home, ".sigstore", "root"))
 	}
 	return rootDir
 }
 
 func cachedRemote(cacheRoot string) string {
-	return filepath.Join(cacheRoot, "remote.json")
+	return filepath.FromSlash(filepath.Join(cacheRoot, "remote.json"))
 }
 
 func cachedTargetsDir(cacheRoot string) string {
-	return filepath.Join(cacheRoot, "targets")
+	return filepath.FromSlash(filepath.Join(cacheRoot, "targets"))
 }
 
-type targetImpl interface {
-	Get(string) ([]byte, error)
-	setImpl
+// Local store implementations
+func newLocalStore() (client.LocalStore, error) {
+	if noCache() {
+		return client.MemoryLocalStore(), nil
+	}
+	tufDB := filepath.FromSlash(filepath.Join(rootCacheDir(), "tuf.db"))
+	local, err := tuf_leveldbstore.FileLocalStore(tufDB)
+	if err != nil {
+		return nil, fmt.Errorf("creating cached local store: %w", err)
+	}
+	return local, nil
 }
 
-type setImpl interface {
-	Set(string, []byte) error
+type localInitFunc func() (client.LocalStore, error)
+
+type embeddedLocalStore struct {
+	// This is where updated repository info will be written to, if needed.
+	writeable     client.LocalStore
+	writeableFunc localInitFunc
 }
 
-type memoryCache struct {
-	targets map[string][]byte
+var GetEmbedded = func() fs.FS {
+	return embeddedRootRepo
 }
 
-func (m *memoryCache) Set(p string, b []byte) error {
-	if m.targets == nil {
-		m.targets = map[string][]byte{}
+func wrapEmbeddedLocal(s localInitFunc) client.LocalStore {
+	return &embeddedLocalStore{writeableFunc: s, writeable: nil}
+}
+
+func (e embeddedLocalStore) GetMeta() (map[string]json.RawMessage, error) {
+	if e.writeable != nil {
+		// We are using a writeable store, so use that.
+		return e.writeable.GetMeta()
+	}
+	// We haven't needed to create or write new metadata, so get the embedded metadata.
+	meta := make(map[string]json.RawMessage)
+	ed, ok := GetEmbedded().(fs.ReadDirFS)
+	if !ok {
+		return nil, errors.New("fs.ReadDirFS unimplemented for embedded repo")
+	}
+	entries, err := ed.ReadDir("repository")
+	if err != nil {
+		return nil, err
+	}
+	rd, ok := GetEmbedded().(fs.ReadFileFS)
+	if !ok {
+		return nil, errors.New("fs.ReadFileFS unimplemented for embedded repo")
+	}
+	for _, entry := range entries {
+		if entry.Type().IsRegular() {
```

Comment:
Because this is the metadata store, I need to avoid copying over the target directory -- that would have `ModeDir` set. Also just in case there's executables or something funky. I like your skipping paradigm!